addons/website/models/ir_http.py

```diff
@@ -100,7 +99,7 @@ def _get_languages(cls):
     @classmethod
     def _get_language_codes(cls):
         if getattr(request, 'website', False):
-            return request.website._get_languages()
+            return request.website._get_cached('_get_languages')
```

Comment:
Please, check the original patch:
 - https://github.com/Vauxoo/odoo/pull/355

Since that the original one is patching the method `_get_default_lang` instead

Reply:
But... we need to cache `_get_languages` method too since that it taking a important result in the python profile

Reply:
Fixed.